### PR TITLE
Fix label fatboot

### PR DESCRIPTION
**Pull request by TheRealFalcon**
Continuation of #500.

Proposed squashed commit message:
```
Recognize LABEL_FATBOOT labels

Update DataSourceNoCloud and ds-identify to recognize LABEL_FATBOOT labels from blkid.
Also updated associated tests.

LP: #1841466
```

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
Github's diff view here is a little unfortunate. Beyond incorporating @smoser 's suggested changes, I added a class around the pre-existing pytest tests, added an additional non-BSD test (since we didn't have any), and added an extra `FATBOOT_LABEL=...` assert to each of the tests. The tests did find an uninitialized variable error, so I fixed that as well.

@smoser and @igalic , since you commented on the last one, I thought you might be interested in my updates here too.

File: tests/unittests/test_util.py [hunk 1]
```diff
@@ -967,96 +967,125 @@ def test_get_proc_ppid(self):
...
+
+        devlist = util.find_devs_with_freebsd(criteria="TYPE=iso9660")
+        assert devlist == ['/dev/iso9660/config-2']
+
+        devlist = util.find_devs_with_freebsd(criteria="TYPE=vfat")
+        assert devlist == ['/dev/msdosfs/EFISYS']
+
+        # lp: #1841466
+        devlist = util.find_devs_with_freebsd(criteria="LABEL_FATBOOT=A_LABEL")
+        assert devlist == []
```
Comment (igalic) [inline tests/unittests/test_util.py:1029]: has anyone tried booting a freebsd image under these conditions to see what the result looks like, and what we could be looking for?

Comment (TheRealFalcon) [reply tests/unittests/test_util.py:1029]: @igalic Was that one of the BSD people you pinged on the last ticket? Should I ping them here too? If we make an effort to reach out but they're unavailable, do you think the unit test changes are sufficient?

Comment (igalic) [reply tests/unittests/test_util.py:1029]: We have roughly 3 *BSD people currently active:

- @goneri
- @ad-m
- @igalic

of those, @goneri is the most experienced when it comes to NoCloud

Comment (goneri) [reply tests/unittests/test_util.py:1029]: Aloha,

Thank you @TheRealFalcon and @igalic, I will give a try to the patch.

Comment (OddBloke) [inline tests/unittests/test_util.py:1029]: It'd be nice to see these tests parameterised.

File: tests/unittests/test_util.py [hunk 2]
```diff
@@ -967,96 +967,125 @@ def test_get_proc_ppid(self):
...
+        assert devlist == ["/dev/cd0"]
+
+        m_subp.side_effect = side_effect_values
+        devlist = util.find_devs_with_netbsd(criteria="TYPE=vfat")
+        assert devlist == ["/dev/ld0", "/dev/dk0", "/dev/dk1"]
+
+        # lp: #1841466
+        m_subp.side_effect = side_effect_values
+        devlist = util.find_devs_with_netbsd(criteria="LABEL_FATBOOT=A_LABEL")
+        assert devlist == ['/dev/ld0', '/dev/dk0', '/dev/dk1', '/dev/cd0']
```
Comment (OddBloke) [inline tests/unittests/test_util.py:1089]: These too.

**Pull request merged**